core/kernel/symbols.c

```diff
@@ -0,0 +1,105 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2019, EPAM Systems
+ */
+
+#include <stdio.h>
+#include <symbols.h>
+#include <util.h>
+
+/* All extern entries are auto-generated during first and second linking pass*/
+
+/* Symbol table */
+extern const struct syms_table syms_table[] __weak;
```

Comment:
Yes, because those symbols are defined in other translation unit.

Reply:
what surprise me here is both `extern` and `__weak` on the same declaration.

Reply:
Yes... `extern` is needed because this symbol is defined in `syms_data.c.[01]`. And we need `__weak`, because during first linking pass those symbols are undefined, so linker will fail with `undefined symbol` error. 

`syms_data.c.0` will be generated only after the first linker pass.

Reply:
Checkpatch is not happy with `extern` in C source. Maybe declare them from `symbols.h`.